book/src/configuration.md

```diff
@@ -1 +1,76 @@
 # Configuration
+
+## Theme
+
+Use a custom theme by placing a theme.toml in your config directory (i.e ~/.config/helix/theme.toml). The default theme.toml can be found [here](https://github.com/helix-editor/helix/blob/master/theme.toml), and user submitted themes [here](https://github.com/helix-editor/helix/blob/master/contrib/themes).
+
+Styles in theme.toml are specified of in the form:
+
+```toml
+"name" = { fg = "#ffffff", bg = "#000000", modifiers = ["bold", "italic"] }
+```
+ where name represents what you want to style, fg specifies the foreground colour, bg the background colour, and modifiers is a list of style modifiers. bg or bold can be omitted to defer to the defaults.
+
+ If you only want to specify the foreground color, you can do so as:
+
+ ```toml
+ "name" = "#ffffff"
+ ```
+
+ Possible entries for modifiers are:
+
+| modifier |
+------------
+| bold |
+| dim |
+| italic |
+| underlined |
+| slow\_blink |
+| rapid\_blink |
+| reversed |
+| hidden |
+| crossed\_out |
+
+ Possible entries for name are:
+
+| name | notes |
+----------------
+| attribute | |
+| keyword | |
+| keyword.directive | preprocessor directives (\#if in C) |
+| namespace | |
+| punctuation | |
+| punctuation.delimiter | |
+| operator | |
+| special | |
+| property | |
+| variable | |
+| variable.parameter | |
+| type | |
+| type.builtin | |
+| constructor | |
+| function | |
```

Comment:
Another thing to note, this matches tree-sitter scopes. https://tree-sitter.github.io/tree-sitter/syntax-highlighting#theme

For a given highlight produced, styling will be determined based on the longest matching theme key.

We sorta half-follow the common scopes from here with some differences: https://macromates.com/manual/en/language_grammars

So it's enough to provide `function` to highlight `function.macro` and `function.builtin` as well, but you can use more specific scopes to highlight specific cases differently.